### PR TITLE
enhance(sync): selection export scope for pods v2

### PR DESCRIPTION
**Pull request by Harshita-mindfire**
This PR aims to support selection scope while exporting a note with export pod v2. This also benefits the `Copy As` command. 

# Dendron Extended PR Checklist

- NOTE: the links don't work. you'll need to go into the wiki and use lookup to find the note until we fix some issues in the markdown export

### Basics

- [x] code should follow [Code Conventions](dev.process.code)
- [x] circular dependency check: make sure your code is not introducing new circular dependencies in plugin-core.  See [Avoiding Circular Dependencies](dev.process.code.best-practices).
- [x] sticking to existing conventions instead of creating new ones
  - eg: [if configuration for utilities are already in one module or package, add future utilities there as well](https://github.com/dendronhq/dendron/pull/1960#discussion_r786228021)

### Extended
- General
  - [x] check whether code be simplified
  - [x] check if similar function already exist in the codebase. if so, can it be re-used?
  - [x] check if this change adversely impact performance
- Operations
  - [x] when shipping this change, will it just work or will it introduce additional operational overhead due to complicated interface or known bugs?
- Architecture
  - [x] check if code is introducing changes on a foundational class or interface. if so, call for design review if needed
    - eg: [making changes to DNode](https://github.com/dendronhq/dendron/pull/2158#pullrequestreview-854689586)

### Basics
- [~] if you are adding analytics related changes, make sure the [Telemetry](https://wiki.dendron.so/notes/84df871b-9442-42fd-b4c3-0024e35b5f3c.html) docs are updated

### Extended
- [~] can we track the performance of this change to know if it is _successful_?
  - eg: [see usage for export pod](https://github.com/dendronhq/dendron/pull/2190#pullrequestreview-855715612)

### Basics

- [x] [Write Tests](dev.process.qa.test) 
- [x] [Confirm existing tests pass](dev.process.qa.test)
- [x] [Confirm manual testing](dev.process.qa.test) 
- [x] Common cases tested
- [ ] 1-2 Edge cases tested
- [~] If your tests changes an existing snapshot, snapshots have been [updated](dev.process.qa.test)

## Docs
- [ ] if your change reflects documentation changes, also submit a PR to [dendron-site](https://github.com/dendronhq/dendron-site) and mention the doc PR link in your current PR
- [~] does this change introduce a new or better way of doing things that others need to be aware of? if so, an async should be created and a process added in [Development](dev) or [Packages](pkg)

### Extended
- [~]  is this a developer BREAKING change? if another person cloning from this branch will need to adjust their dependencies or mental model of the architecture, then it is. if this is the case, make sure this is communicated according to [Close Loop](dev.process.close-loop)
  - eg: [breaking dev change due to new dependency](https://github.com/dendronhq/dendron/pull/2188#pullrequestreview-855696330)

### CODE REVIEW

File: packages/plugin-core/src/commands/pods/BaseExportPodCommand.ts [hunk 1]
```diff
@@ -283,10 +283,13 @@ export abstract class BaseExportPodCommand<
               const filteredPayload = opts.payload.filter(
                 (note) => note.fname !== savedNote.fname
               );
-              await this.executeExportNotes({
-                ...opts,
-                payload: filteredPayload.concat(savedNote),
-              });
+              // export only selection if present instead of entire body
+              const exportOpts =
```
Comment (kevinslin) [inline packages/plugin-core/src/commands/pods/BaseExportPodCommand.ts:287]: can we raise an warning if nothing is selected?

Comment (Harshita-mindfire) [reply packages/plugin-core/src/commands/pods/BaseExportPodCommand.ts:287]: We don't have a separate scope for `clipboard`. We export the selected text if the export scope is `note` and selection is present. Not very sure if that would be the right call to raise a warning on each note export.

Comment (kevinslin) [reply packages/plugin-core/src/commands/pods/BaseExportPodCommand.ts:287]: can we modify this so we actually create a new scope "selection" and only apply this on selection? 

this way, we can be specific about the selection case. we also don't want to accidentally export the selection when the user meant to export the note.

copy as should then always set the `selection` scope

Comment (Harshita-mindfire) [reply packages/plugin-core/src/commands/pods/BaseExportPodCommand.ts:287]: got it

Comment (Harshita-mindfire) [reply packages/plugin-core/src/commands/pods/BaseExportPodCommand.ts]: [warning toaster](https://github.com/dendronhq/dendron/pull/3619/files#diff-b300d2a5c67b0dd9dd53963dac70b830cd1ae68b29c8de1cc7daeee764cd40bcR409)

**Review by namjul — APPROVED** [review]
just read it through without trying locally and lgtm.

**Comment by Harshita-mindfire** [comment]
docs: https://github.com/dendronhq/dendron-site/pull/655/files

**Review by hikchoi — CHANGES_REQUESTED** [review]
Tried testing this out locally, but it looks like activation silently fails (no errors, did not activate).

File: packages/plugin-core/src/components/pods/PodControls.ts [hunk 2]
```diff
@@ -530,6 +533,9 @@ export class PodUIControls {
       case PodExportScope.Workspace:
         return "Exports all notes in the Dendron workspace";
 
+      case PodExportScope.Selection:
+        return "Export the selection from currently opened note";
```
Comment (jonathanyeung) [inline packages/plugin-core/src/components/pods/PodControls.ts:537]: nit: maybe "selected text" instead of "selection" just to make it clearer.

**Review by jonathanyeung — APPROVED** [review]
re: @hikchoi 's comment - is the init failure unique to this change commit or was it introduced elsewhere?  Otherwise lgtm

**Comment by hikchoi** [comment]
> re: @hikchoi 's comment - is the init failure unique to this change commit or was it introduced elsewhere? Otherwise lgtm

looks like this is because of the (already fixed) regression with `showWhitelistingLocalhostDocsIfNecessary`.

I rebased it locally and tried again and it activated correctly.

**Review by hikchoi — APPROVED** [review]
Just tried it out locally (had to rebase master to fix the activation issue), and looks good. `copy as` and export pod works great!

**Pull request merged**